tests/end-to-end/regression/5884/tests/FooTest.php

```diff
@@ -61,10 +61,11 @@ public function testStreamToNonWritableFileWithPHPUnitErrorHandler(): void
 
         // Now verify the original file is unchanged.
         $contents = file_get_contents($filename);
-        $this->assertSame('foo', $contents);
 
         chmod($filename, 0o755);
         unlink($filename);
+
+        $this->assertSame('foo', $contents);
```

Comment:
whats the background of this change?

Reply:
I'm atm  working on Windows. Running tests inside WSL container throws error when particular test is executed. After debugging why it's happening (WSL host file-system ignores permissions and file is truncated), corrected the test to `unlink` before failing. It generated new temp file for each failure.